### PR TITLE
Interpolation support for Canvas styles

### PR DESCRIPTION
**Pull request by tschaub**
This adds support for using the `interpolate` operator with numbers and colors for flat styles on the Canvas renderer.

The new [populated places example](https://deploy-preview-15141--ol-site.netlify.app/en/latest/examples/populated-places.html) shows how `interpolate` can be used to create a color ramp and scale points.

### CODE REVIEW

File: src/ol/render/canvas/style.js [hunk 1]
```diff
@@ -208,6 +215,21 @@ export function buildStyle(flatStyle, context) {
...
+    !evaluateImage &&
+    !isEmpty(flatStyle)
+  ) {
+    // assume this is a user error
+    // would be nice to check the properties and suggest "did you mean..."
+    throw new Error(
+      'No fill, stroke, point, or text symbolizer properties in style: ' +
+        JSON.stringify(flatStyle)
+    );
+  }
```
Comment (tschaub) [inline src/ol/render/canvas/style.js:231]: I pulled this out as a separate pull request.  See #15142.

File: examples/populated-places.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import Map from '../src/ol/Map.js';
+import TileLayer from '../src/ol/layer/Tile.js';
+import VectorLayer from '../src/ol/layer/Vector.js';
+import VectorSource from '../src/ol/source/Vector.js';
+import View from '../src/ol/View.js';
+import XYZ from '../src/ol/source/XYZ.js';
+
+const populatedPlaces = new VectorLayer({
+  source: new VectorSource({
+    url: 'http://localhost:4000/vector/populated-places.json',
```
Comment (mike-000) [inline examples/populated-places.js:12]: Needs to change to live data link

File: src/ol/style/RegularShape.js [hunk 3]
```diff
@@ -505,6 +505,9 @@ class RegularShape extends ImageStyle {
     this.renderOptions_ = this.createRenderOptions();
     const size = this.renderOptions_.size;
     this.canvas_ = {};
+    if (this.size_ && (this.size_[0] != size || this.size_[1] != size)) {
+      this.hitDetectionCanvas_ = null;
+    }
```
Comment (tschaub) [inline src/ol/style/RegularShape.js:510]: We don't currently properly hit detect circles after a call to `circle.setRadius()`.  I pulled this change out in a separate pull request with tests: #15143.

File: examples/populated-places.js (hunk 2 above)
Comment (tschaub) [reply examples/populated-places.js]: Updated in the latest commit.

File: examples/tooltip-on-hover.js [hunk 4]
```diff
@@ -25,8 +25,6 @@ const map = new Map({
 });
 
 const info = document.getElementById('info');
-info.style.visibility = 'hidden';
-info.style.pointerEvents = 'none';
```
Comment (tschaub) [inline examples/tooltip-on-hover.js:29]: I moved these defaults to the CSS to avoid the tooltip flash at initial render.

**Review by ahocevar — APPROVED** [review]

**Comment by tschaub** [comment]
Thanks for the review, @ahocevar.

**Comment by ahocevar** [comment]
Thanks for the hard work on this, @tschaub!
